examples/advanced/examples/foundry_fork_db.rs

```diff
@@ -0,0 +1,70 @@
+//! Foundry Fork DB
+use std::sync::Arc;
+
+use eyre::Result;
+
+use alloy::{
+    eips::BlockId,
+    network::AnyNetwork,
+    node_bindings::Anvil,
+    primitives::U256,
+    providers::{Provider, ProviderBuilder},
+    rpc::types::BlockTransactionsKind,
+};
+use foundry_fork_db::{cache::BlockchainDbMeta, BlockchainDb, SharedBackend};
+use revm_primitives::{BlockEnv, CfgEnv, Env};
+
+// TODO: Add docs and explanation of the workflow.
+// TODO: Depict how the backend handler can smartly manages duplicate requests.
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let anvil = Anvil::new().spawn();
+    let provider = ProviderBuilder::new().network::<AnyNetwork>().on_http(anvil.endpoint_url());
+
+    let cfg = CfgEnv::default().with_chain_id(31337);
+
+    let block =
+        provider.get_block(BlockId::latest(), BlockTransactionsKind::Hashes).await?.unwrap();
+
+    let pin_block = BlockId::number(block.header.number);
+    let block = BlockEnv {
+        number: U256::from(block.header.number),
+        coinbase: block.header.miner,
+        timestamp: U256::from(block.header.timestamp),
+        gas_limit: U256::from(block.header.gas_limit),
+        blob_excess_gas_and_price: None,
+        difficulty: U256::from(block.header.difficulty),
+        prevrandao: None,
+        basefee: U256::from(block.header.base_fee_per_gas.unwrap()),
+    };
```

Comment:
we should make creating BlockchainDbMeta easier with a helper function that accepts `Block`

tracking